rust/agents/scraper/src/chain_scraper/mod.rs

```diff
@@ -349,38 +245,128 @@ impl SqlChainScraper {
     }
 }
 
-#[derive(Debug, Clone)]
-struct Delivery {
-    message_id: H256,
-    meta: LogMeta,
+#[async_trait]
+impl HyperlaneDB<HyperlaneMessage> for HyperlaneSqlDb {
+    /// Store messages from the origin mailbox into the database.
+    async fn store_logs(&self, messages: &[(HyperlaneMessage, LogMeta)]) -> Result<u32> {
+        if messages.is_empty() {
+            return Ok(0);
+        }
+        let txns: HashMap<H256, TxnWithId> = self
+            .ensure_blocks_and_txns(messages.iter().map(|r| &r.1))
+            .await?
+            .map(|t| (t.hash, t))
+            .collect();
+        let storable = messages.iter().map(|m| {
+            let txn = txns.get(&m.1.transaction_hash).unwrap();
+            StorableMessage {
+                msg: m.0.clone(),
+                meta: &m.1,
+                txn_id: txn.id,
+            }
+        });
+        self.db
+            .store_dispatched_messages(&self.mailbox_address, storable)
+            .await?;
+
+        Ok(messages.len() as u32)
+    }
 }
 
-impl Delivery {
-    fn as_storable(&self, txn_id: i64) -> StorableDelivery {
-        StorableDelivery {
-            message_id: self.message_id,
-            meta: &self.meta,
-            txn_id,
+#[async_trait]
+impl HyperlaneDB<H256> for HyperlaneSqlDb {
```

Comment:
Seeing `impl HyperlaneDB<H256>` does not make it immediately clear it is storing message deliveries.

Reply:
What would you recommend? type MessageID = H256?

Reply:
I think a (private?) type of `type DeliveredMessageId = H256` would work.

Reply:
I did the former